modules/creating-manifest-file-customized-br-ex-bridge.adoc

```diff
@@ -0,0 +1,185 @@
+
+:FeatureName: Creating a `MachineConfig` object that includes a customized `br-ex` bridge
+include::snippets/technology-preview.adoc[]
+endif::postinstall-bare-metal-ipi,postinstall-bare-metal-upi[]
+
+ifdef::postinstall-bare-metal-ipi,postinstall-bare-metal-upi[]
+As an alternative to using the `configure-ovs.sh` shell script to set a customized `br-ex` bridge on a bare-metal platform, create a `NodeNetworkConfigurationPolicy` object that includes a customized `br-ex` bridge network configuration.
+
+:FeatureName: Creating a ``NodeNetworkConfigurationPolicy` object that includes a customized `br-ex` bridge
+include::snippets/technology-preview.adoc[]
+
+This feature supports the following tasks:
+
+* Modifying the maximum transmission unit (MTU) for your cluster.
+* Rollback your Kubernetes OVN plugin configuration to an earlier state.
+* Switch to a different bond interface, such as MIImon (Media Independent Interface Monitor), bonding mode, or Quality of Service (QoS).
+* Updating DNS values.
+endif::postinstall-bare-metal-ipi,postinstall-bare-metal-upi[]
+
+Consider the following use cases for creating manifest object that includes a customized `br-ex` bridge:
+
+* You want to make postinstallation changes to the bridge, such as changing the Open vSwitch (OVS) or OVN-Kubernetes `br-ex` bridge network. The `configure-ovs.sh` shell script does not support making postinstallation changes to the bridge.
+* You want to deploy the bridge on a different interface than the interface available on a host or server IP address.
+* You want to make advanced configurations to the bridge that are not possible with the `configure-ovs.sh` shell script. Using the script for these configurations might result in the bridge failing to connect multiple network interfaces and facilitating data forwarding between the interfaces.
+
+ifndef::postinstall-bare-metal-ipi,postinstall-bare-metal-upi[]
+
+[NOTE]
+====
+If you require an environment with a single network interface controller (NIC) and default network settings, use the `configure-ovs.sh` shell script.
+====
+
+After you install {op-system-first} and the system reboots, the Machine Config Operator injects Ignition configuration files into each node in your cluster, so that each node received the `br-ex` bridge network configuration. The Operator prevents `br-ex` bridge configuration conflicts by sending a signal to the `configure-ovs.sh` shell script to not configure the `br-ex` bridge.
+endif::postinstall-bare-metal-ipi,postinstall-bare-metal-upi[]
+
+.Prerequisites
+
+
+.Procedure
+ifndef::postinstall-bare-metal-ipi,postinstall-bare-metal-upi[]
+. Create a `MachineConfig` manifest file and define a customized `br-ex` bridge network configuration similiar to the following example:
++
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
```

Comment:
Let me just give an example of what I'm looking for with this, because this still isn't quite it:

```
kind: MachineConfig
metadata:
  labels:
    machineconfiguration.openshift.io/role: worker
  name: 10-br-ex-worker
spec:
  config:
    ignition:
      version: 3.2.0
    storage:
      files:
      - contents:
          source: data:text/plain;charset=utf-8;base64,<base64_encoded_nmstate_configuration>
        mode: 0644
        overwrite: true
        path: /etc/nmstate/openshift/cluster.yml
```

and the unencoded contents of that encoded NMState field would look like this:

```
interfaces:
- name: enp2s0
  type: ethernet
  state: up
  ipv4:
    enabled: false
  ipv6:
    enabled: false
- name: br-ex
  type: ovs-bridge
 [snipped, but the entire rest of the example should be here]
```

These need to be completely separate because the unencoded value can't be used directly in the machine-config manifest, but the content of that encoded value is hugely important and we need it included in the example.